### PR TITLE
Fix bug where ScrollContainer could cause layout engine to loop forever.

### PR DESCRIPTION
**Pull request by eoineoineoin**
Copy-pasting repro from https://github.com/space-wizards/space-station-14/issues/13641

> To repro (there's a slighly simpler way to repro it, I'm sure):
> 1. Write a paper with the contents "11111 11111i 11111ii 11111iii 11111iiii 11111iiiii 11111iiiiii 11111iiiiiii 11111iiiiiiii 11111iiiiiiiii 11111iiiiiiiiii 11111iiiiiiiiiii 11111iiiiiiiiiiii 11111iiiiiiiiiiiii 11111iiiiiiiiiiiiii 11111iiiiiiiiiiiiiii 11111iiiiiiiiiiiiiiii 11111iiiiiiiiiiiiiiiii"
> 2. Resize the paper so the entered text just fills up the the whole page
    Hang forever

What's happening is that the ScrollContainer and RichTextLabel don't get on. The ScrollContainer decides it needs to enable the scrollbar, which causes an InvalidateMeasure(), then, on next layout, the RichTextLabel decides it needs to add a line break to account for the slightly more narrow space. Then, on next layout, the scroll bar isn't necessary anymore, which causes an InvalidateMeasure() and....

ScrollContainer _attempts_ to account for the size of the scroll bars, but when they are `Visible==false`, their `DesiredSize` is 0. This adds the ability for ScrollBars to be Visible for the purposes of layout, but won't draw themselves if not necessary. Longer-term, this feels like it would be a nice to break Visibility out into `Invisible` `Invisible-but-takes-up-space` and `Visible`.

I tested this on a bunch of the other UIs which use ScrollContainer, and they seem to be behaving correctly.

### CODE REVIEW

**Review by PJB3005 — CHANGES_REQUESTED** [review]
This should just be a new boolean property on `ScrollBar`, not making `Visible` virtual.

File: Robust.Client/UserInterface/Control.cs [hunk 1]
```diff
@@ -207,6 +210,33 @@ private void _propagateVisibilityChanged(bool newVisible)
...
         }
 
+        /// <summary>
+        ///     Whether or not this control and its children require
+        ///     space to be reserved, even when not visible.
+        /// </summary>
+        /// <seealso cref="ReservesSpace"/>
+        [ViewVariables(VVAccess.ReadWrite)]
+        [Animatable]
+        public virtual bool ReservesSpace
```
Comment (PJB3005) [inline Robust.Client/UserInterface/Control.cs]: Remove `virtual` here.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Minor nit otherwise looks good.

File: Robust.Client/UserInterface/Control.cs (hunk 1 above)
Comment (eoineoineoin) [reply Robust.Client/UserInterface/Control.cs]: Whoops, my bad; sorry.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**